From 61f2d2f7e308c42cce652db4a172cfa4b0ff6bf1 Mon Sep 17 00:00:00 2001
From: OPNA2608 <opna2608@protonmail.com>
Date: Sat, 18 Oct 2025 22:45:37 +0200
Subject: [PATCH] tests/testhash.c: Properly request shutdown of test2 threads

So they can be destroyed properly. Fixes sometimes-occuring SIGSEGVs.
---
 tests/testhash.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/tests/testhash.c b/tests/testhash.c
index 0769aa6..cb6ed24 100644
--- a/tests/testhash.c
+++ b/tests/testhash.c
@@ -134,7 +134,12 @@ int test2(void)
 	}
 
 	for (i = 0; i < ttl; i++) {
-		ks_thread_destroy(&threads[i]);
+		ks_thread_request_stop(threads[i]);
+	}
+
+	for (i = 0; i < ttl; i++) {
+		ks_thread_join(threads[i]);
+		if (ks_thread_destroy(&threads[i]) != KS_STATUS_SUCCESS) return 0;
 	}
 
 
-- 
2.51.0

